### PR TITLE
Add backend functions in explainer_manager.py to handle dashboard requests

### PR DESCRIPTION
**Pull request by gaugup**
## Description

Add function `request_explanations` to `explainer_manager.py` to handle explanations requests from RAI dashboard

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: responsibleai/responsibleai/managers/explainer_manager.py [hunk 1]
```diff
@@ -179,9 +179,54 @@ def get_data(self):
...
+            raise UserConfigValidationException(
+                'Only one row of data is allowed for '
+                'local explanation generation.')
+
+        if self._classes is not None:
+            model_task = ModelTask.Classification
+        else:
+            model_task = ModelTask.Regression
+
+        explainer = MimicExplainer(
```
Comment (imatiach-msft) [inline responsibleai/responsibleai/managers/explainer_manager.py:213]: nit: Can this be refactored with the mimic explainer created in the compute method into one common method?  It looks like the code in both places is the same.  Similarly the explain_global method, and probably the new explain_local addition, can be refactored out into the common method.

File: responsibleai/tests/explainer/test_explainer_manager.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            test=X_test.iloc[0:10],
+            target_column='target',
+            task_type='classification')
+        rai_insights.explainer.add()
+        rai_insights.compute()
+
+        global_explanations = \
+            rai_insights.explainer.request_explanations(
+                local=False, data=X_test.drop(['target'], axis=1).iloc[0:10])
+        assert global_explanations is not None
```
Comment (imatiach-msft) [inline responsibleai/tests/explainer/test_explainer_manager.py:33]: nit: can these asserts be refactored out into a separate method and called in both tests?  It looks like we are validating the same properties in both test cases on global and local explanations.

**Review by imatiach-msft — APPROVED** [review]

File: responsibleai/responsibleai/managers/explainer_manager.py [hunk 3]
```diff
@@ -179,9 +179,54 @@ def get_data(self):
...
         """
         return [
-            self._get_interpret(i) for i in self.get()]
+            self._get_interpret(
+                i, self._evaluation_examples) for i in self.get()]
+
+    def request_explanations(self, local: bool, data: Any):
+        """Return the explanations for a given point(s) .
+
+        :param local: True if local explanations are quested
```
Comment (imatiach-msft) [inline responsibleai/responsibleai/managers/explainer_manager.py:188]: are quested -> are requested

File: responsibleai/responsibleai/managers/explainer_manager.py [hunk 4]
```diff
@@ -179,9 +179,54 @@ def get_data(self):
...
+                      and False otherwise.
+        :type local: bool
+        :param data: The data point(s) for which the explanations
+                     need to be generated.
+        :type data: Any
+        :return: The serialized version explanations for the
```
Comment (imatiach-msft) [inline responsibleai/responsibleai/managers/explainer_manager.py:194]: nit: remove version, eg
The serialized explanations for the given data point(s).

File: responsibleai/responsibleai/managers/explainer_manager.py [hunk 5]
```diff
@@ -179,9 +179,54 @@ def get_data(self):
...
+                 given data point(s).
+        :rtype: Any
```
Comment (imatiach-msft) [inline responsibleai/responsibleai/managers/explainer_manager.py:196]: I think there is some BaseExplanation type that can be used here to be more specific than Any

File: responsibleai/responsibleai/managers/explainer_manager.py (hunk 1 above)
Comment (gaugup) [reply responsibleai/responsibleai/managers/explainer_manager.py]: Done.

File: responsibleai/tests/explainer/test_explainer_manager.py (hunk 2 above)
Comment (gaugup) [reply responsibleai/tests/explainer/test_explainer_manager.py]: Done.

File: responsibleai/responsibleai/managers/explainer_manager.py (hunk 3 above)
Comment (gaugup) [reply responsibleai/responsibleai/managers/explainer_manager.py]: Done

File: responsibleai/responsibleai/managers/explainer_manager.py (hunk 4 above)
Comment (gaugup) [reply responsibleai/responsibleai/managers/explainer_manager.py]: Done.

File: responsibleai/responsibleai/managers/explainer_manager.py (hunk 5 above)
Comment (gaugup) [reply responsibleai/responsibleai/managers/explainer_manager.py]: Changed it to be ModelExplanationData.

**Pull request merged**